s3fs/core.py

```diff
@@ -1042,6 +1042,9 @@ def walk(self, path, maxdepth=None, **kwargs):
 
     def modified(self, path, version_id=None, refresh=False):
         info = self.info(path=path, version_id=version_id, refresh=refresh)
+        if 'LastModified' not in info:
+            # This path is a bucket or folder, which do not currently have a modified date
+            return
```

Comment:
Seems a little odd to just ignore. Could raise `IsADirectoryError`?

Reply:
Yep - no strong opinion here @martindurant, let me add one.